pkg/clientcache/cache.go

```diff
@@ -112,19 +112,25 @@ func (c *ClientCache) CreateRemoteClientsFromSecret(namespacedName types.Namespa
 
 // GetRestConfig takes the ClientConfig and parses the *rest.Config from it
 func (c *ClientCache) GetRestConfig(assistCfg *api.ClientConfig) (*rest.Config, error) {
-	apiConfig, err := c.extractClientCmdFromSecret(types.NamespacedName{Namespace: assistCfg.Namespace, Name: assistCfg.Spec.KubeConfigSecret.Name})
+	secretKey := types.NamespacedName{Namespace: assistCfg.Namespace, Name: assistCfg.Spec.KubeConfigSecret.Name}
+	apiConfig, err := c.extractClientCmdApiConfigFromSecret(secretKey)
 	if err != nil {
 		return nil, err
 	}
 
-	clientCfg := clientcmd.NewNonInteractiveClientConfig(*apiConfig, assistCfg.Name, &clientcmd.ConfigOverrides{}, nil)
-	return clientCfg.ClientConfig()
+	clientCmdCfg := clientcmd.NewNonInteractiveClientConfig(
+		*apiConfig,
+		assistCfg.GetContextName(),
```

Comment:
I think it's more correct to call `assistCfg.GetContextName()` instead of accessing `assistCfg.Name`.

Reply:
Agreed